commit adb013e2cb9e79e3f7be0bc7d013d1eb4510a5e5
Author: Chris Patterson <pattersonc@ainfosec.com>
Date:   Mon Feb 9 16:13:05 2015 -0500

    qemu-device: add disk readonly option and switch to scsi
    
    Signed-off-by: Chris Patterson <pattersonc@ainfosec.com>

diff --git a/src/qemu-device.c b/src/qemu-device.c
index 68f1437..da3c83b 100644
--- a/src/qemu-device.c
+++ b/src/qemu-device.c
@@ -224,21 +224,25 @@ static bool drive_device_parse_options (struct device_model *devmodel,
     char *media;
     char *format;
     char *index;
+    char *readonly;
     bool res = false;
 
     file = retrieve_option (devmodel, device, "file", drivefile);
     media = retrieve_option (devmodel, device, "media", drivemedia);
     format = retrieve_option (devmodel, device, "format", driveformat);
     index = retrieve_option (devmodel, device, "index", driveindex);
+    readonly = retrieve_option (devmodel, device, "readonly", drivereadonly);
 
     SPAWN_ADD_ARGL (devmodel, end_drive, "-drive");
     SPAWN_ADD_ARGL (devmodel, end_drive,
-                    "file=%s,if=ide,index=%s,media=%s,format=%s",
-                    file, index, media, format);
+                    "file=%s,if=scsi,index=%s,media=%s,format=%s,readonly=%s",
+                    file, index, media, format, readonly);
 
     res = true;
 
 end_drive:
+    free (readonly);
+drivereadonly:
     free (index);
 driveindex:
     free (format);
